packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/connector/ConnectorContext.tsx

```diff
@@ -19,14 +19,19 @@ import {
   ConnectorContextValue,
 } from './ConnectorContext.types';
 
-export const ConnectorContext = React.createContext<ConnectorContextValue>(undefined);
+const defaultValue: ConnectorContextValue = {
+  connection: null,
+  setConnection: () => {},
+};
+
+export const ConnectorContext = React.createContext<ConnectorContextValue>(defaultValue);
 
 export const ConnectorContextProvider = ({ children }: ConnectorContextProviderProps) => {
   const [state, setState] = useState<ConnectorContextProviderState>({
     connection: null,
   });
 
-  const setConnection = (connection: Connection) => setState((prevState) => ({ ...prevState, connection }));
+  const setConnection = (connection: Connection | null) => setState((prevState) => ({ ...prevState, connection }));
```

Comment:
Can the connection really be `null`?

Reply:
For now yes, to reset the connection on contextual menu close.
See `useConnector#onConnectorContextualMenuClose`.
I will provide a new function to the context to reset the connection.